metadata-ingestion/src/datahub/ingestion/source/neo4j/neo4j_source.py

```diff
@@ -0,0 +1,350 @@
+                    lastModified=AuditStampClass(
+                        time=round(time.time() * 1000),
+                        actor="urn:li:corpuser:ingestion",
+                    ),
+                    fields=fields,
+                ),
+                systemMetadata=DatasetPropertiesClass(
+                    customProperties={"properties": "property on object"}
+                ),
+            )
+            self.report.obj_created += 1
+            return mcp
+        except Exception as e:
+            log.error(e)
+            self.report.obj_failures += 1
+
+    def add_tag_to_dataset(
+        self, table_name: str, tag_name: str
+    ) -> MetadataChangeProposalWrapper:
+        graph = DataHubGraph(
+            DatahubClientConfig(server=self.ctx.pipeline_config.sink.config["server"])
+        )
+        dataset_urn = make_dataset_urn(
+            platform=self.config.platform, name=table_name, env=self.config.environment
+        )
+        current_tags: Optional[GlobalTagsClass] = graph.get_aspect(
+            entity_urn=dataset_urn,
+            aspect_type=GlobalTagsClass,
+        )
+        tag_to_add = make_tag_urn(tag_name)
+        tag_association_to_add = TagAssociationClass(tag=tag_to_add)
+
+        if current_tags:
+            if tag_to_add not in [x.tag for x in current_tags.tags]:
+                current_tags.tags.append(TagAssociationClass(tag_to_add))
+        else:
+            current_tags = GlobalTagsClass(tags=[tag_association_to_add])
+        return MetadataChangeProposalWrapper(
+            entityUrn=dataset_urn,
+            aspect=current_tags,
+        )
+
+    def get_neo4j_metadata(self, query: str) -> pd.DataFrame:
+        driver = GraphDatabase.driver(
+            self.config.uri, auth=(self.config.username, self.config.password)
+        )
+        """
+        This process retrieves the metadata for Neo4j objects using an APOC query, which returns a dictionary
+        with two columns: key and value. The key represents the Neo4j object, while the value contains the
+        corresponding metadata.
+
+        When data is returned from Neo4j, much of the relationship metadata is stored with the relevant node's
+        metadata. Consequently, the objects are organized into two separate dataframes: one for nodes and one for
+        relationships.
+
+        In the node dataframe, several fields are extracted and added as new columns. Similarly, in the relationship
+        dataframe, certain fields are parsed out, while others require metadata from the nodes dataframe.
+
+        Once the data is parsed and these two dataframes are created, we combine a subset of their columns into a
+        single dataframe, which will be used to create the DataHub objects.
+
+        See the docs for examples of metadata:  metadata-ingestion/docs/sources/neo4j/neo4j.md
+        """
+        log.info(f"{query}")
+        with driver.session() as session:
+            result = session.run(query)
```

Comment:
Can you please add `try..except` here and then use `report.failure` in case this fails to execute due to some reason ?

Reply:
try except added